lib/govuk_navigation_helpers/navigation_sidebar.rb

```diff
@@ -0,0 +1,85 @@
+require 'govuk_navigation_helpers/services'
+require 'govuk_navigation_helpers/configuration'
+
+module GovukNavigationHelpers
+  class NavigationSidebar
+    def initialize(content_item)
+      @content_item = ContentItem.new content_item
+    end
+
+    def navigation_sidebar
+      {
+        related_items: related_items,
+        collections: related_collections,
+        topics: related_topics,
+        policies: related_policies,
+        publishers: related_organisations,
+        other: [related_external_links, related_contacts]
+      }
+    end
+
+  private
+
+    def process_links_for_sidebar(related_content)
+      related_content.map do |link|
+        {
+          text: link["title"],
+          path: link["base_path"]
+        }
+      end
+    end
+
+    def related_items
+      quick_links = @content_item.quick_links.map do |link|
```

Comment:
It might be more uniform to have a method which returns pre-built quick links and pre-built `@content_item.navigation_sidebar_related_links` and have this method simply return the appropriate set based on the `quick_links.any?` condition. Not a blocker though.